packages/cli/medusa-cli/src/commands/new.ts

```diff
@@ -169,7 +169,7 @@ const copy = async (starterPath, rootPath) => {
 }
 
 // Clones starter from URI.
-const clone = async (hostInfo, rootPath, v2 = false) => {
+const clone = async (hostInfo, rootPath, v2 = false, ci = false) => {
```

Comment:
thought: Would it make sense to be able to pass a repo URL (with a branch) instead? That would allow us to eliminate all these conditionals further down. And we won't have to change the command in the future.

Reply:
I think it's possible. We would instead just supply a `branch` option and use it

Reply:
@olivermrbl changed it to a `branch` option

Reply:
Think something like this could work. Fine to move forward with this. 

What I had in mind was actually something like this: `medusa new ci https://github.com/medusajs/medusa-starter-default/[branch]`

But this is what I am unsure if we can do without having to do too many gymnastics around extracting the branch from the url.

Reply:
Makes sense, actually I think it would be nice to re-do the entire CLI tool because there are a lot of things that could be fixed/improved, but a task for another day 😄